src/hardware/voodoo.cpp

```diff
@@ -119,38 +119,31 @@ class VOODOO : public Module_base {
         }
     }
 
-    void PCI_InitEnable(Bitu val) {
-        switch (emulation_type) {
-            case EmulationType::Software:
-            case EmulationType::OpenGL:
-                Voodoo_PCI_InitEnable(val);
-                break;
-            default:
-                break;
-        }
-    }
+	template <typename Func>
+	inline void ExecuteForEmulationType(EmulationType emulation_type, Func&& func) {
+		switch (emulation_type) {
+			case EmulationType::Software:
+			case EmulationType::OpenGL:
+				func();
+				break;
+			default:
+				break;
+		}
```

Comment:
This is very strange - seeing functions passed into other functions, only for a single call to be made.

When I look at all these nearly empty functions and nearly empty case statements, it tells me a lot of simplification can happen.

Reply:
Should I just remove the template?

Reply:
@kcgen This is the last thing still is unresolved... No clue what you recommend me to do here